extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/ReactiveMongoCollection.java

```diff
@@ -478,7 +502,7 @@
      * @param pipeline the aggregate pipeline
      * @return a stream containing the result of the aggregation operation
      */
-    AggregatePublisher<Document> aggregateAsPublisher(List<? extends Bson> pipeline);
+    AggregatePublisher<T> aggregateAsPublisher(List<? extends Bson> pipeline);
```

Comment:
I'm not sure if this should be considered as a breaking change. 
Initially, it only contained `Document`, now it can be any type. 

First I would use `TDocument` as in the driver to indicate that it's not any type, but must be somehow a document. The lack of constraint in the driver is a bit odd.

Reply:
@cescoffier `TDocument` means any type `T` that can be converted to a document. I agree this is a bit odd as it didn't convey any constraint. As we register the Pojo Codec, almost any type can be converted automatically (except for some deep generic types). So maybe `T` is OK.